plugins/web-terminal/src/components/TerminalComponent/TerminalComponent.tsx

```diff
@@ -128,7 +166,29 @@ export const TerminalComponent = () => {
             type="password"
             variant="outlined"
             inputRef={tokenRef}
+            required
           />
+          <Select
+            name="namespace"
+            inputRef={namespaceRef}
+            defaultValue={defaultNamespace}
```

Comment:
Why to set `defaultNamespace` as a default value if we've just verified it can't be used by the user?

Reply:
Makes sense I think it is better to have some default value so that users will not be trying to make a request with undefined namespace (also `select` component doesn't like it very much, I would have to add this empty value (or any other placeholder) as a value to `namespaces`), so I have changed it to first value in array `namespaces` this way it will at least try request with valid namespace if the user would just enter token and cliked enter...

Reply:
Well, exactly this case is covered in MUI documentation. Try it.

https://v4.mui.com/components/selects/#select


Mark the `Select` as required (you did that in the last iteration) and include an arbitrary menu item with empty value `""` before the `namespaces.map()`. This will fail client-side validation.